source/postprocess/visualization.cc

```diff
@@ -326,11 +330,13 @@ namespace aspect
       if (std::isnan(last_output_time))
         {
           last_output_time = this->get_time() - output_interval;
+          last_output_timestep = this->get_timestep_number();
         }
 
       // return if graphical output is not requested at this time
       if ((this->get_time() < last_output_time + output_interval)
-          && (this->get_timestep_number() != 0))
+          && (this->get_timestep_number() != 0)
+	  && (this->get_timestep_number() < last_output_timestep + maximum_timesteps_between_outputs))
```

Comment:
Forgive me for being nitpicky, but could you change the order of the conditions? This would be a bit simpler to read if it would be:
```
if ((time < output_time)
&& (step < output_steps)
&& (timestep != 0))
```
Also make sure the `&&` align. With so complex code we learned how important it is to keep things as easy to read as possible.